### PR TITLE
[NEW] Display preferences screen

### LINKED ISSUE #3259: wrong icon for "unread on top" on the sorting bar
### Description:
on the sorting panel the "unread on top" item has an eye icon and there should be a flag icon

### Environment Information:

- Rocket.Chat Server Version: 3.15.1
- Rocket.Chat App Version: 4.17.0.24389
- Device Name:  Iphone 11 pro max
- OS Version: ios 14.6

### Steps to reproduce:

1. open sort panel
2. "unread on top" has an eye icon (screenshot 1)
3. but this is an icon to "hide" (screenshot 2)
4. "unread..." there should be a "flag" (screenshot 3)

### Expected behavior:

2. "unread on top" has an flag icon

### Actual behavior:

2. "unread on top" has an eye icon

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

Modify how the user changes the display preferences as expanded or condensed, avatar hide/show. And the sort preferences, like sort by activity or name, unread on top, group types.
## Issue(s)	

Feature to modify the room list's display, adding this, new way to sort the room list
Closes #3259

## How to test or reproduce

- Open the Side Bar
- Tap on Display
- Change something
- And back to Room List

## Screenshots

https://user-images.githubusercontent.com/47038980/134728179-5e70750f-0afb-4219-bb9a-8521e66c719f.mp4

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/DisplayPrefsView.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+	);
+};
+
+DisplayPrefsView.propTypes = {
+	theme: PropTypes.string,
+	navigation: PropTypes.object,
+	isMasterDetail: PropTypes.bool
+};
+
+export default (withTheme(DisplayPrefsView));
```
Comment (gerzonc) [inline app/views/DisplayPrefsView.js:211]: Use `useTheme` hook instead

File: app/views/DisplayPrefsView.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+		/>
+	);
+
+	return (
+		<SafeAreaView testID='status-view'>
+			<StatusBar />
+			<List.Container testID='display-view-list'>
+				<List.Section>
+					<List.Separator />
+					<List.Item
```
Comment (gerzonc) [inline app/views/DisplayPrefsView.js:124]: Title should be `bold` as well

**Review by gerzonc — CHANGES_REQUESTED** [review]

File: app/views/DisplayPrefsView.js (hunk 2 above)
Comment (gerzonc) [reply app/views/DisplayPrefsView.js:136]: Use the shared styles for bold

File: app/views/DisplayPrefsView.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+const styles = StyleSheet.create({
+	title: {
+		flexShrink: 1,
+		fontSize: 16,
+		...sharedStyles.textMedium
+	}
+});
+
+
+const DisplayPrefsView = (props) => {
```
Comment (gerzonc) [inline app/views/DisplayPrefsView.js:31]: Changes need to be reflected correctly on `TeamChannelsView` as well

File: app/views/DisplayPrefsView.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+						)}
+					/>
+					<List.Item
+						left={() => <List.Icon name='view-extended' />}
+						title='Expanded'
+						testID='expanded-display-pref'
+						right={() => renderRadio(displayType === 'expanded')}
+						onPress={displayExpanded}
+					/>
+					<List.Separator />
```
Comment (gerzonc) [inline app/views/DisplayPrefsView.js:152]: You should add a `List.Separator` at the beginning and the end of each `List.Section`

File: app/views/RoomsListView/index.js [hunk 5]
```diff
@@ -141,7 +138,9 @@ class RoomsListView extends React.Component {
 		insets: PropTypes.object,
 		queueSize: PropTypes.number,
 		inquiryEnabled: PropTypes.bool,
-		encryptionBanner: PropTypes.string
+		encryptionBanner: PropTypes.string,
+		showAvatar: PropTypes.bool,
+		displayType: PropTypes.string
```
Comment (gerzonc) [inline app/views/RoomsListView/index.js:143]: Depending on the `displayType` you should modify as well the alignment for the content that appears on swipe, so we prevent this:
image

File: app/views/RoomsListView/index.js [hunk 6]
```diff
@@ -84,11 +82,12 @@ const shouldUpdateProps = [
 	'searchText',
 	'loadingServer',
 	'showServerDropdown',
-	'showSortDropdown',
 	'sortBy',
 	'groupByType',
 	'showFavorites',
 	'showUnread',
+	'showAvatar',
+	'displayType',
```
Comment (gerzonc) [inline app/views/RoomsListView/index.js:90]: For some reason it takes a couple of seconds (10-15) to update the `RoomsListView` after updating the sorting preference, might be related to how this screen is being updated

**Review by gerzonc — CHANGES_REQUESTED** [review]

File: app/views/RoomsListView/index.js (hunk 6 above)
Comment (reinaldonetof) [reply app/views/RoomsListView/index.js]: https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3318/commits/d6d460b37b43b1caff7b15e02e77c86e0d93fd57

**Review by gerzonc — APPROVED** [review]
LGTM

File: app/views/DisplayPrefsView.js [hunk 7]
```diff
@@ -0,0 +1,210 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Switch } from 'react-native';
+
```
Comment (diegolmello) [inline app/views/DisplayPrefsView.js:4]: Unnecessary empty space.

File: app/views/DisplayPrefsView.js [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+import { RadioButton } from 'react-native-ui-lib';
+import { useDispatch, useSelector } from 'react-redux';
+import { setPreference } from '../actions/sortPreferences';
```
Comment (diegolmello) [inline app/views/DisplayPrefsView.js:7]: Add empty space

File: app/views/DisplayPrefsView.js [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+			) : (
+				<HeaderButton.Drawer navigation={navigation} testID='display-view-drawer' />
+			))
+		});
+	}, []);
+
+	const setSortPreference = (param) => {
+		try {
+			dispatch(setPreference(param));
+			RocketChat.saveSortPreference(param);
```
Comment (diegolmello) [inline app/views/DisplayPrefsView.js:50]: Can we await this?

File: app/i18n/locales/en.json [hunk 10]
```diff
@@ -772,5 +772,13 @@
   "Converting_Team_To_Channel": "Converting Team to Channel",
   "Select_Team_Channels_To_Delete": "Select the Team’s Channels you would like to delete, the ones you do not select will be moved to the Workspace. \n\nNotice that public Channels will be public and visible to everyone.",
   "You_are_converting_the_team": "You are converting this Team to a Channel",
+  "Display_Preference": "Display Preference",
+  "Display": "Display",
```
Comment (diegolmello) [inline app/i18n/locales/en.json:776]: We need to check other files and remove the strings we're not using anymore.

File: app/views/RoomsListView/index.js [hunk 11]
```diff
@@ -98,6 +91,19 @@ const shouldUpdateProps = [
...
+	'showFavorites',
+	'showUnread'
+];
+
+const displayPropsShouldUpdate = [
+	'showAvatar',
+	'displayMode'
+];
+
 const getItemLayout = (data, index) => ({
```
Comment (diegolmello) [inline app/views/RoomsListView/index.js:107]: We should apply to condensed as well.

**Review by diegolmello — CHANGES_REQUESTED** [review]
Almost there :)

**Review by AlexAlexandre — APPROVED** [review]
LGTM :)

**Pull request merged**